---
manual:Javascript
version:0
lang:zh
rawUrl:https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/URIError/prototype
---



这篇翻译不完整。请帮忙从英语[翻译这篇文章]46468 "")。




`**URIError.prototype**`属性表示[`URIError`]40565 "URIError 对象用来表示以一种错误的方式使用全局URI处理函数而产生的错误。")构造器的原型。

## 描述<a name="描述"></a>


所有的[`URIError`]40565 "URIError 对象用来表示以一种错误的方式使用全局URI处理函数而产生的错误。")实例都继承自`URIError.prototype`。 可以通过原型(prototype) 给所有的实例添加属性或者方法。


## 属性<a name="属性"></a>
<dl><dt id=''>`URIError.prototype.constructor`</dt><dd>声明创建实例原型 (prototype) 的方法。</dd><dt id=''>[`URIError.prototype.message`]41987 "message 属性是有关错误信息，人类易读的（human-readable）描述。")</dt><dd>错误信息。虽然 ECMA-262 规范指出[`URIError`]40565 "URIError 对象用来表示以一种错误的方式使用全局URI处理函数而产生的错误。")应该提供其自己专属的`**message**`****属性，但是在[SpiderMonkey]42011 "")中，该属性继承自[`Error.prototype.message`]41987 "message 属性是有关错误信息，人类易读的（human-readable）描述。")</dd><dt id=''>[`URIError.prototype.name`]41988 "name 属性表示error类型的名称.初始值为"Error".")</dt><dd>错误名称。继承自[`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")。</dd><dt id=''>[`URIError.prototype.fileName`]41985 "fileName 属性包含引发此错误的文件的路径.")</dt><dd>产生该错误的代码所在文件的路径。 继承自[`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")。</dd><dt id=''>[`URIError.prototype.lineNumber`]41986 "lineNumber 属性的值为抛出错误的代码在其源文件中所在的行号。")</dt><dd>产生该错误的代码所在行的行号。继承自[`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")。</dd><dt id=''>[`URIError.prototype.columnNumber`]41984 "columnNumber属性包含引发此错误的文件行中的列号。")</dt><dd>产生该错误的代码所在列的列号。 继承自[`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")。</dd><dt id=''>[`URIError.prototype.stack`]42012 "此页面仍未被本地化, 期待您的翻译!")</dt><dd>堆栈记录。继承自[`Error`]10494 "通过Error的构造器可以创建一个错误对象。当运行时错误产生时，Error的实例对象会被抛出。Error对象也可用于用户自定义的异常的基础对象。下面列出了各种内建的标准错误类型。")。</dd></dl>
## 方法<a name="方法"></a>


虽然[`URIError`]40565 "URIError 对象用来表示以一种错误的方式使用全局URI处理函数而产生的错误。")的原型对象自身不包含任何方法，但是[`URIError`]40565 "URIError 对象用来表示以一种错误的方式使用全局URI处理函数而产生的错误。")的实例通过原型链(prototype chain)继承了一些方法。


## 规范<a name="规范"></a>

Specification | Status | Comment 
[ECMAScript 3rd Edition (ECMA-262)<br></br><small>NativeError.prototype</small>]46465 "") | Standard | 初始定义 
[ECMAScript 5.1 (ECMA-262)<br></br><small>NativeError.prototype</small>]45987 "") | Standard | 定义为`*NativeError*.prototype`. 
[ECMAScript 2015 (6th Edition, ECMA-262)<br></br><small>NativeError.prototype</small>]45988 "") | Standard | `*NativeError*.prototype`. 
[ECMAScript Latest Draft (ECMA-262)<br></br><small>NativeError.prototype</small>]45989 "") | Draft | `*NativeError*.prototype`. 


## 浏览器兼容性<a name="浏览器兼容性"></a>
[新的兼容性表格正在测试中<i></i>]3360 "")

 | <abbr>Desktop<i></i></abbr> | <abbr>Mobile<i></i></abbr> | <abbr>Server<i></i></abbr> 
 | <abbr>Chrome<i></i></abbr> | <abbr>Edge<i></i></abbr> | <abbr>Firefox<i></i></abbr> | <abbr>Internet Explorer<i></i></abbr> | <abbr>Opera<i></i></abbr> | <abbr>Safari<i></i></abbr> | <abbr>Android webview<i></i></abbr> | <abbr>Chrome for Android<i></i></abbr> | <abbr>Edge Mobile<i></i></abbr> | <abbr>Firefox for Android<i></i></abbr> | <abbr>Opera for Android<i></i></abbr> | <abbr>iOS Safari<i></i></abbr> | <abbr>Samsung Internet<i></i></abbr> | <abbr>Node.js<i></i></abbr> 
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  | 
[Basic support]46469 "") | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>1 | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>4 | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes | <abbr>Full support</abbr>Yes 


### Legend<a name="Legend"></a>
<dl><dt id=''><abbr>Full support</abbr></dt><dd>Full support</dd></dl>


## 相关链接<a name="相关链接"></a>

* [`Error.prototype`]41983 "所有 Error 与 非标准Error 的实例都继承自 Error.prototype。同所有构造器函数一样，你可以在构造器的 prototype 上添加属性或者方法，使其在所有该构造器的实例上生效。")
* [`Function.prototype`]41691 "Function.prototype 属性存储了 Function 的原型对象。")



